Added ms5837 files

jessetham opened the pull request.

rummanwaqar commented on README.md:
```diff
@@ -7,4 +7,17 @@ The parameter file stores the publisher name.
+`roslaunch au_sensors ms5837.launch`
+
+Node expects device to be at:
+`/dev/arvp/depth_sensor`
```
Create a parameter file similar to the one for OS5000. This will save things like port url so we don't have to rebuild if port changes. I will also store the abs pressure and density values.

rummanwaqar commented on include/au_sensors/ms5837.h:
```diff
@@ -0,0 +1,102 @@
+/*
+ * Rumman Waqar
```
Add your name here

rummanwaqar commented on include/au_sensors/ms5837.h:
```diff
@@ -0,0 +1,102 @@
+	 */
+	void filter_callback( std::string token );
+
+	/*
```
I will implement a checksum in depth sensor embedded system so we can ensure the data we get is correct. Then you can make use of it

rummanwaqar commented on launch/ms5837.launch:
```diff
@@ -0,0 +1,9 @@
+		output="screen"
+		respawn="true">
+	</node>
+</launch>
```
Look at the sensors.launch file and add ms5837 to it so we can start both sensors with one command.

rummanwaqar commented on src/ms5837.cpp:
```diff
@@ -0,0 +1,86 @@
+	buffer_.wait_and_pop( line );
+	
+	/*
+	// check length
```
get rid of all these comments. 
We will check the checksum so keep it here until we implement it. I will be done implementing it on the embedded device in the next 30 mins.

rummanwaqar commented on src/ms5837_node.cpp:
```diff
@@ -0,0 +1,95 @@
+	
+	// ROS Publishers
+	ros::Publisher pressure_pub_;
+	ros::Publisher depth_pub_;
```
we will have a single publisher for both pressure and depth

rummanwaqar commented on src/ms5837_node.cpp:
```diff
@@ -0,0 +1,95 @@
+	pressure_pub_ = private_nh_.advertise<au_core::Depth>("pressure", 10);
+	ROS_INFO("Publishing pressure data on /ms5837/pressure");
+	depth_pub_ = private_nh_.advertise<au_core::Depth>("depth", 10);
+	ROS_INFO("Publishing depth data on /ms5837/depth");
```
single publisher

rummanwaqar commented on src/ms5837_node.cpp:
```diff
@@ -0,0 +1,95 @@
+
+	// Initialize the sensor
+	try {	
+		depth_sensor_ = new MS5837("/dev/arvp/depth_sensor", 115200);
```
The port name will be a param. Look at the os5000_node for implementation exmaples.

rummanwaqar commented on src/ms5837_node.cpp:
```diff
@@ -0,0 +1,95 @@
+		depth_pub_.publish(msg);
+	}
+
+	ros::spinOnce();
```
move spinOnce command to the while loop in DepthSensor::spin() under the update command.

rummanwaqar commented on src/ms5837_node.cpp:
```diff
@@ -0,0 +1,95 @@
+void DepthSensor::spin() {
+	while(ros::ok()) {
+		update();
+	}
```
ros::spinOnce() command gets moved here,

rummanwaqar commented:
I have added some modifications. When they are done, add a comment with a list of changes. :+1: cheers

jessetham commented:
**Commit de65f51**
- Added a parameter file for the ms5837 node
- Added ms5837 to sensors.launch
- Edited the ms5837 node to only have one publisher
- Changed the parsing format to match the new Teensy serial output
- Implemented checksum
